app/controllers/session_controller.rb

```diff
@@ -90,46 +123,44 @@ def create_session(user, password)
   end
 
   def redirect_after_sucessful_login
-    if session[:jumpto].blank?
-      redirect_to search_path
-    else
-      jump_to = session[:jumpto]
-      session[:jumpto] = nil
-      redirect_to jump_to
-    end
+    jump_to = session[:jumpto] || search_path
+    session[:jumpto] = nil
+    redirect_to jump_to
   end
 
-  def set_session_attributes(user, password)
+  def set_session_attributes(user, pk_secret)
     jumpto = session[:jumpto]
     reset_session
     session[:jumpto] = jumpto
     session[:username] = user.username
     session[:user_id] = user.id.to_s
-    session[:private_key] = user.decrypt_private_key(password)
+    session[:private_key] = user.decrypt_private_key(pk_secret)
     session[:last_login_at] = user.last_login_at
     session[:last_login_from] = user.last_login_from
   end
 
   def password_params_valid?
-    unless current_user.authenticate(params[:old_password])
+    unless current_user.authenticate_db(params[:old_password])
```

Comment:
make sure this can only be used if user is db user and not API user !
early return? exception ?